opm/simulators/linalg/MILU.hpp

```diff
@@ -86,14 +106,14 @@ struct RealReorderer : public Reorderer
 
   
 template<class M>
-void milu0_decomposition(M& A, std::function<double(const double&)> absFunctor = AuxFunctors::signFunctor(),
-			  std::function<double(const double&)> signFunctor = AuxFunctors::oneFunctor(),
+void milu0_decomposition(M& A, double absFunctor(const double&) = signFunctor<Opm::detail<double>>(const double&),
+			 double signFunctor(const double&) = oneFunctor<Opm::detail<double>>(const double&),
```

Comment:
This seems a bit messed up. You want to pass it the signFunctor function in the Opm::detail namespace as default arguments.  As you are in ` Opm::detail` the namespace can be omitted here. Please be aware that `Opm::detail<double>` is wrong. It may be double, but more correct would be to use `typename M::field_type` instead. Also to refer to a function you probably may omit the brackets with function parameters but they should not hurt either

But the type of the function parameter does not change it is still a function that gets a double and returns a double like before, hence `std::function<double(const double&)> `.

Reply:
New version std::function are back! So far no errors but compilation hasn't finished yet
Thakns!
sorry for the many confusing pieces of code :